feature_engine/encoding/count_frequency.py

```diff
@@ -143,16 +148,44 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
 
         y: pandas Series, default = None
             y is not needed in this encoder. You can pass y or None.
+
+        threshold : list of floats
```

Comment:
I would make threshold take 1 value, either int if encoding method is count, or a float if encoding method is frequency. And we would apply the same threshold to all variables.

If the user wants different thresholds for different variable groups, then they can instantiate the transformer twice or more times with the different thresholds and variable groups.

Otherwise the class becomes too complicated.

This docstring should be moved to the init. And in the init, we need to add a catch, if int and count ok, otherwise error, if float and frequency ok, otherwise error. 


the default parameter of threshold is None, which means no categories would be grouped.

Reply:
Got it, Ill do the changes and resubmit.